crates/wasmtime/src/module.rs

```diff
@@ -347,6 +347,12 @@ impl Module {
             }
         };
 
+        if engine.config().mlock_module_mmaps {
+            for (mmap, _info) in artifacts.iter() {
+                mmap.mlock()?;
+            }
+        }
```

Comment:
I think that if this loop were moved to [`Module::from_parts`](https://github.com/bytecodealliance/wasmtime/blob/8070f142dc96e00871052f296451ff595f6a4d09/crates/wasmtime/src/module.rs#L544-L629) it should catch all cases of module construciton and avoid the need to special case `deserialize_file`